### PR TITLE
Fix negative compilation tests in glsl-conformance-test.js.

### PR DESCRIPTION
**Pull request by kenrussell**
Addresses regressions in the following tests:
  conformance/glsl/misc/boolean_precision.html
  conformance2/glsl3/switch-case.html
  conformance2/glsl3/texture-offset-non-constant-offset.html
  conformance2/glsl3/uniform-block-layouts.html

Follow-up to #3215 .

### CODE REVIEW

**Comment by kenrussell** [comment]
Blocking roll of WebGL conformance suite into Chromium under http://crbug.com/1165751 / https://chromium-review.googlesource.com/2690219 . Taking the liberty of merging myself.

**Comment by jdashg** [comment]
Hey so I understand that because of how it's currently implemented, you end up in situations like this, but this PR should really not block that roll. You need a way to choose non-tip for import.

File: sdk/tests/js/glsl-conformance-test.js [hunk 1]
```diff
@@ -162,8 +162,10 @@ function runOneTest(gl, info) {
       }
     }
     // Save the shaders so we test shared shader.
-    if (vShader) {
-      vShaderDB[vSource] = vShader;
+    if (compiledVShader) {
+      vShaderDB[vSource] = compiledVShader;
+    } else {
+      vShader = null;
```
Comment (jdashg) [inline sdk/tests/js/glsl-conformance-test.js:168]: Use compiledVShader wherever vShader is used below instead. The meaning of variables should avoid changing over the course of a function.

Comment (kenrussell) [reply sdk/tests/js/glsl-conformance-test.js:168]: compiledVShader and compiledFShader are scoped within their blocks and are inaccessible to the code below. Feel free to refactor further as long as the tests continue to pass.

**Pull request merged**